File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @param {number} threshold
 * @return {number}
 */
var numOfSubarrays = function(arr, k, threshold) {
  const target = threshold * k;
  let sum = arr.slice(0, k).reduce((a, b) => a + b);
  let ans = +(sum >= target);
  
  for (let i = k; i < arr.length; i++) {
    sum += arr[i] - arr[i - k];
    if (sum >= target) ans++;
  }
  
  return ans;
};

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        target = threshold * k
        s = sum(arr[:k])
        ans = +(s >= target)
        for i in range(k, len(arr)):
            s += arr[i] - arr[i - k]
            ans += +(s >= target)
        return ans